hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java

```diff
@@ -170,6 +171,22 @@ public InetAddress getRemoteAddress() throws IOException {
     }
   }
 
+  /**
+   * Return String created from OMRequest userInfo. If userInfo is not
+   * set, returns null.
+   * @return String
+   * @throws IOException
+   */
+  @VisibleForTesting
+  public String getHostName() throws IOException {
+    if (omRequest.hasUserInfo()) {
+      return InetAddress.getByName(omRequest.getUserInfo()
```

Comment:
Here why do we need this, we can direcly return hostName which we have in UserInfo
omRequest.getUserInfo().getHostName() right? Not sure if i am missing something here.

Reply:
This will return the cannonical host name. Say you say -- host1--> this call will resolve it with DNS or hosts and return the full host name. Another use case, which might not be applicable for us is that the name might be short name or IP address. This ends up looking the right name from the network layer. That is all; I will hold off on committing this since it is night for @cxorm .

Reply:
It do think in our case, getHostname from userInfo might suffix too. I think we do this due diligence already

Reply:
Because we set the hostName information in preExecute by calling remoteAddress.getHostName(), and in ValidateAndPreExecuteCache, we don't need to call again is my point here.

Reply:
yes, agree.

Reply:
Thanks @bharatviswa504 for the comment.
After a little time going through, I also think this line can be cleaner.
We're going to fix related method and UT.

Reply:
Thanks for the update @cxorm 
In getHostName, we don't need InetAddress.getByName, we can directly use the hostName if it is has, we can use it.

Reply:
Implementation of getHostName in this commit is correct.
https://github.com/apache/hadoop-ozone/pull/70/commits/015ba21ea84d75586d6c21e2d23a984334db7ae3

Reply:
Thanks @bharatviswa504 for the review.
I haven't complete the patch : )

Yes the implementation of getHostName in patch 015ba21 is correct, but the corresponding UT ```TestOMClientRequestWithUserInfo``` has a failure on L119 ```Assert.assertEquals(inetAddress.getHostName(), hostName);```.

I'am going to find the cause that why the L116 can work and L119 fail.

Reply:
We found that we should mock the method ```ProtobufRpcEngine.Server#getRemoteIp``` in ```TestOMClientRequestWithUserInfo```, so that the ```OMClientRequest#getUserInfo``` can set related value in ```userInfo```.

Fixed.